### PR TITLE
[ci skip] test_all_sandia: add caraway for Hip with AMD GPU testing

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

File: scripts/testing_scripts/test_all_sandia [hunk 1]
```diff
@@ -495,6 +499,21 @@ elif [ "$MACHINE" = "mayer" ]; then
...
+
+  HIPCLANG_BUILD_LIST="Hip_Serial"
+  HIPCLANG_WARNING_FLAGS=""
+
+  # Format: (compiler module-list build-list exe-name warning-flag)
+  COMPILERS=("rocm/4.2.0 $BASE_MODULE_LIST $HIPCLANG_BUILD_LIST hipcc $HIPCLANG_WARNING_FLAGS")
+
+  if [ -z "$ARCH_FLAG" ]; then
+    ARCH_FLAG="--arch=VEGA906"
+  fi
```
Comment (masterleinad) [inline scripts/testing_scripts/test_all_sandia:515]: You should not need that if the machine has access to the GPU, i.e. if you are not cross compiling.

**Review by masterleinad — APPROVED** [review]
Looks good to me but I don't have access to the machine to check. :slightly_smiling_face:

File: scripts/testing_scripts/test_all_sandia [hunk 2]
```diff
@@ -495,6 +499,21 @@ elif [ "$MACHINE" = "mayer" ]; then
     ARCH_FLAG="--arch=ARMV8_THUNDERX2"
   fi
 
+elif [ "$MACHINE" = "caraway" ]; then
+  SKIP_HWLOC=True
+
+  BASE_MODULE_LIST="cmake/3.19.3,<COMPILER_NAME>/<COMPILER_VERSION>"
+
+  HIPCLANG_BUILD_LIST="Hip_Serial"
+  HIPCLANG_WARNING_FLAGS=""
```
Comment (dalg24) [inline scripts/testing_scripts/test_all_sandia:508]: why are they no warning flags?

Comment (ndellingwood) [reply scripts/testing_scripts/test_all_sandia]: I added warning flags to match those from the Jenkins CI jobs, SHA ac76a3054105bb05a59b53a35cbcb38f5037cf7e

**Review by dalg24 — APPROVED** [review]

**Pull request merged**